esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -1814,17 +1814,28 @@ bool CFileSprayEx::onGetDFUExceptions(IEspContext &context, IEspGetDFUExceptions
     return true;
 }
 
-void CFileSprayEx::readAndCheckSpraySourceReq(MemoryBuffer& srcxml, const char* srcIP, const char* srcPath,
+void CFileSprayEx::readAndCheckSpraySourceReq(MemoryBuffer& srcxml, const char* srcIP, const char* srcPath, const char* srcPlane,
     StringBuffer& sourceIPReq, StringBuffer& sourcePathReq)
 {
     StringBuffer sourcePath(srcPath);
-    sourceIPReq.set(srcIP);
-    sourceIPReq.trim();
     sourcePath.trim();
+
     if(srcxml.length() == 0)
     {
+        if (!isEmptyString(srcPlane))
+        {
+            Owned<IPropertyTree> plane =  getStoragePlane(srcPlane);
+            if (!plane)
+                throw MakeStringException(ECLWATCH_INVALID_INPUT, "Unknown storage plane: %s", srcPlane);
+            const char * storagePlanePath = plane->queryProp("@prefix");
+            if (!startsWith(srcPath,storagePlanePath))
+                throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid source path");
+            sourceIPReq.set("127.0.0.1"); // with storage planes, all locations are local
```

Comment:
Unfortunately not true... we also support bare-metal drop zones (on cloud and bare-metal).  See https://github.com/hpcc-systems/HPCC-Platform/pull/15038

Reply:
What should sourceIPReq be set to? If it's empty, how should RemoteMultiFilename be handled (ws_fsService.cpp lines 1943-1948)?

Reply:
See the linked PR for accessing the ip if it is non localhost